tests/Flywheel/FlywheelTest.js

```diff
@@ -580,7 +580,7 @@ describe('Flywheel', () => {
       const tx = await send(comptroller, 'claimComp', [a2]);
       const a2AccruedPost = await compAccrued(comptroller, a2);
       const compBalancePost = await compBalance(comptroller, a2);
-      expect(tx.gasUsed).toBeLessThan(400000);
+      expect(tx.gasUsed).toBeLessThan(500000);
```

Comment:
😬

Reply:
Yeah, it's not great. We can either require borrowers to poke their borrow after these changes are made, or take the uninitialized borrow states into account which increases the gas cost. Either way costs gas.

There are some optimizations that can be made to both distributeBorrowerComp and distributeSupplierComp - have a very simple check if the user is a borrower/supplier in the market. If they aren't, no need to proceed with the rest of the function logic if we know they aren't going to be accruing COMP in that [side of the] market.

These optimizations aren't included in this merge request as many people are waiting on the features and bug fixes within this branch. I don't want to delay this any longer.

In the meantime, it's best to update the UI to call claim on the markets the user is active in, rather than the generic claim function which iterates all markets. Even after optimizations are made, this approach is best. The cost of claim grows with each additional market on Compound, so it's best for the UI to utilize view functions to make claims for the user as efficient as possible.